dashboard/permissions.py

```diff
@@ -0,0 +1,39 @@
+"""
+Permission classes for the dashboard
+"""
+from django.contrib.auth.models import User
+from django.http import Http404
+from rest_framework.generics import get_object_or_404
+from rest_framework.permissions import BasePermission
+
+from roles.roles import (
+    Instructor,
+    Staff,
+)
+
+
+class CanReadIfStaffOrSelf(BasePermission):
+    """
+    Only staff on a program the learner is enrolled in can
+    see their dashboard. Learners can view their own dashboard.
+    """
+
+    def has_permission(self, request, view):
+        if request.user.is_anonymous():
+            raise Http404
+
+        user = get_object_or_404(User, username=view.kwargs['username'])
```

Comment:
Actually you should also filter on the provider:

    User.objects.filter(social_auth_uid='a_username_here', social_auth__provider=EdxOrgOAuth2.name)